vortex-io/src/object_store.rs

```diff
@@ -46,16 +50,23 @@ impl VortexReadAt for ObjectStoreReadAt {
         // ourselves to avoid a second copy to align the buffer. Instead, we can write directly
         // into the aligned buffer.
         let mut buffer = ByteBufferMut::with_capacity_aligned(len, alignment);
-
-        let response = object_store
-            .get_opts(
-                &location,
-                GetOptions {
-                    range: Some(GetRange::Bounded(start..end)),
-                    ..Default::default()
-                },
-            )
-            .await?;
+        let io_dispatcher = IO_DISPATCHER.with(|io| io.clone());
+
+        let response = io_dispatcher
+            .dispatch(move || async move {
+                object_store
+                    .get_opts(
+                        &location,
+                        GetOptions {
+                            range: Some(GetRange::Bounded(start..end)),
+                            ..Default::default()
+                        },
+                    )
+                    .await
+            })
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
+            .await
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))??;
 
         let buffer = match response.payload {
```

Comment:
Replace spawn_blocking to use I/O dispatcher?... Hold up... then where is the perf coming from? Not spawning all the exec tasks maybe?

Reply:
looking into it in #2332